examples/nlp/question-answering.py

```diff
@@ -0,0 +1,422 @@
+"""
+Title: Question Answering with Transformers
+Author: Matthew Carrigan and Merve Noyan
+Date created: 13/01/2022
+Last modified: 13/01/2022
+Description: Question answering implementation using Keras and 🤗Transformers.
+"""
+"""shell
+!pip install git+https://github.com/huggingface/transformers.git
+!pip install datasets
+!pip install huggingface-hub
+"""
+
+"""
+# Fine-tuning a model on a question answering task
+"""
+
+"""
+Question answering is a common NLP task with several variants. In some variants, the task
+is multiple-choice:
+A list of possible answers are supplied with each question, and the model simply needs to
+return a probability distribution over the options. A more challenging variant of
+question answering, which is more applicable to real-life tasks, is when the options are
+not provided. Instead, the model is given an input document and a question about the
+document, and it must extract the span of text in the document that contains the answer.
+In this case, the model is not computing a probability distribution over answers, but two
+probability distributions over the tokens in the document text, representing the start
+and end of the span containing the answer.
+
+Question answering of this kind is a very challenging NLP task, and the dataset size
+required to train such a model from scratch when the questions and answers are natural
+language is prohibitively huge. As a result, question answering (like almost all NLP
+tasks) benefits enormously from starting from a strong pre-trained foundation model -
+starting from a strong pre-trained language model can reduce the dataset size required to
+reach a given accuracy by multiple orders of magnitude, enabling you to reach very strong
+performance with surprisingly reasonable datasets.
+
+Starting with a pre-trained model adds difficulties, though - where do you get the model
+from? How do you ensure that your input data is preprocessed and tokenized the same way
+as the original model? How do you modify the model to add an output head that matches
+your task of interest?
+
+In this example, we'll show you how to load a model from the Hugging Face [🤗
+Transformers](https://github.com/huggingface/transformers) library to tackle this
```

Comment:
Make sure to follow the style guide with regard to things like markdown link formatting, etc. https://github.com/keras-team/keras-io/blob/master/contributor_guide.md